packages/react-ui-core/src/Gmap/Markers.js

```diff
@@ -0,0 +1,83 @@
+import { Component } from 'react'
+import PropTypes from 'prop-types'
+import { setupMarker, removeMarker } from './utils/markerHelpers'
+
+const EMPTY_MARKERS = {}
+
+const MARKER = ({ properties, geometry }) => ({
+  id: properties.id,
+  position: {
+    lat: geometry.coordinates[0],
+    lng: geometry.coordinates[1],
+  },
+  title: properties.id,
+  key: properties.id,
+})
+
+export default class Markers extends Component {
+  static propTypes = {
+    map: PropTypes.object,
+    marker: PropTypes.func,
+    geojson: PropTypes.object,
+    append: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    marker: () => ({}),
+    append: false,
+  }
+
+  constructor(props) {
+    super(props)
+    this.markers = EMPTY_MARKERS
+  }
+
+  componentWillUnmount() {
+    this.clearMarkers()
+  }
+
+  clearMarkers() {
+    Object.keys(this.markers).map(id => {
+      removeMarker(this.markers[id])
+      return null
+    })
+
+    this.markers = EMPTY_MARKERS
+  }
+
+  marker(feature) {
+    const props = this.props.marker(feature)
+    const defaults = MARKER(feature)
+
+    if (typeof props === 'object') {
+      return {
+        ...defaults,
```

Comment:
not a big deal but since you only use defaults if props is an object, you could do

```
{
   ...(MARKER(feature)),
   ...props,
}
```

Reply:
That would probably be cleaner 👍

Reply:
oh, forget what i said. i see now defaults is used if the first condition fails